rubin_sim/data/rs_download_data.py

```diff
@@ -95,98 +81,11 @@ def rs_download_data():
-            return 0
-        else:
-            print("Versions do not match")
-            return 1
-
-    if not args.orbits:
-        dirs = [key for key in dirs if "orbits_precompute" not in key]
-
-    # See if base URL is alive
-    url_base = args.url_base
-    try:
-        r = requests.get(url_base)
-        if r.status_code != requests.codes.ok:
-            url_base = BACKUP_DATA_URL
-    except ConnectionError:
-        url_base = BACKUP_DATA_URL
-        print(f"Could not connect to {args.url_base}; trying {url_base}")
-    try:
-        r = requests.get(url_base)
-        fail_message = f"Could not connect to {args.url_base} or {url_base}. Check sites are up?"
-    except ConnectionError:
-        print(fail_message)
-        exit()
-    if r.status_code != requests.codes.ok:
-        print(fail_message)
-        exit()
-
-    for key in dirs:
-        filename = files[key]
-        path = os.path.join(data_dir, key)
-        if os.path.isdir(path) and not args.force:
-            warnings.warn("Directory %s already exists, skipping download" % path)
-        else:
-            if os.path.isdir(path) and args.force:
-                rmtree(path)
-                warnings.warn("Removed existing directory %s, downloading new copy" % path)
-            # Download file
-            url = url_base + filename
-            print("Downloading file: %s" % url)
-            # Stream and write in chunks (avoid large memory usage)
-            r = requests.get(url, stream=True)
-            file_size = int(r.headers.get("Content-Length", 0))
-            if file_size < 245:
-                warnings.warn(f"{url} file size unexpectedly small.")
-            # Download this size chunk at a time; reasonable guess
-            block_size = 1024 * 1024
-            progress_bar = tqdm(total=file_size, unit="iB", unit_scale=True, disable=args.tdqm_disable)
-            print(f"Writing to {os.path.join(data_dir, filename)}")
-            with open(os.path.join(data_dir, filename), "wb") as f:
-                for chunk in r.iter_content(chunk_size=block_size):
-                    progress_bar.update(len(chunk))
-                    f.write(chunk)
-            progress_bar.close()
-            # untar in place
-            unpack_archive(os.path.join(data_dir, filename), data_dir)
-            os.remove(os.path.join(data_dir, filename))
-            versions[key] = files[key]
-
-    # Write out the new version info to the data directory
-    with open(version_file, "w") as f:
-        for key in versions:
-            print(key + "," + versions[key], file=f)
-
-    # Write a little table to stdout
-    new_versions = data_versions()
-    print("Current/updated data versions:")
-    for k in new_versions:
-        if len(k) <= 10:
-            sep = "\t\t"
-        else:
-            sep = "\t"
-        print(f"{k}{sep}{new_versions[k]}")
+    download_rubin_data(
+        files,
+        dirs=args.dirs,
+        print_versions_only=args.versions,
```

Comment:
This looks good, but since I just added the --upgrade flag (and it's really useful for workflows), would you mind adding it here as well? 
(I could always do this in a separate PR too).